src/components/Caption/Caption.jsx

```diff
@@ -0,0 +1,60 @@
+import "./Caption.css";
+import CaptionElements from "./CaptionElements";
+import React, { useState } from "react";
+
+function Caption(props) {
+  const [cbValue, setCbValue] = useState("");
+  const [penClicked, setPenClicked] = useState(false);
+  const [currentCaptionValue, setCurrentCaptionValue] = useState(
+    props.children
+  );
+  const [lastCaptionValue, setLastCaptionValue] = useState(props.children);
```

Comment:
You're passing the value here from the parent component. So it's expected that once you 'save' your changes - the changes are applied not just on this component level, but for the source data as well.